src/Cake.Core/Tooling/ToolResolutionStrategy.cs

```diff
@@ -147,7 +147,16 @@ private List<DirectoryPath> GetPathDirectories()
             {
                 var separator = new[] { _environment.Platform.IsUnix() ? ':' : ';' };
                 var paths = path.Split(separator, StringSplitOptions.RemoveEmptyEntries);
-                result.AddRange(paths.Select(p => new DirectoryPath(p)));
+                foreach (var p in paths)
+                {
+                    try
+                    {
+                        result.Add(new DirectoryPath(p.TrimStart(' ', '"', '\'').TrimEnd(' ', '"', '\'')));
```

Comment:
If we Trim both Start & End can't we just [Trim(chars)](https://msdn.microsoft.com/en-us/library/d4tt83f9.aspx)?